README.md

```diff
@@ -146,18 +146,18 @@ generated for every new singleton type.
 A class used to pass singleton values implicitly. The `sing` method produces
 an explicit singleton value.
 
-    data SomeSing (kproxy :: KProxy k) where
-      SomeSing :: Sing (a :: k) -> SomeSing ('KProxy :: KProxy k)
+    data SomeSing k where
+      SomeSing :: Sing (a :: k) -> SomeSing k
 
 The `SomeSing` type wraps up an _existentially-quantified_ singleton. Note that
 the type parameter `a` does not appear in the `SomeSing` type. Thus, this type
 can be used when you have a singleton, but you don't know at compile time what
-it will be. `SomeSing ('KProxy :: KProxy Thing)` is isomorphic to `Thing`.
+it will be. `SomeSing Thing` is isomorphic to `Thing`.
 
-    class (kparam ~ 'KProxy) => SingKind (kparam :: KProxy k) where
-      type DemoteRep kparam :: *
-      fromSing :: Sing (a :: k) -> DemoteRep kparam
-      toSing   :: DemoteRep kparam -> SomeSing kparam
+    class SingKind k where
+      type DemoteRep k :: *
+      fromSing :: Sing (a :: k) -> DemoteRep k
+      toSing   :: DemoteRep k -> SomeSing k
 
 This class is used to convert a singleton value back to a value in the
 original, unrefined ADT. The `fromSing` method converts, say, a
```

Comment:
Further down, this says:

>  The `DemoteRep` associated
> kind-indexed type family maps a proxy of the kind `Nat`
> back to the type `Nat`.

This is no longer true, but I'm not sure what I should change this to. Should I just say "maps `Nat` to `Nat`"? I don't know if that really elucidates what the purpose of `DemoteRep` is. Maybe we ought to give an example where `DemoteRep k /= k`? (Speaking of which, should the default definition of `DemoteRep` be `DemoteRep k = k`?)